pkg/lexer/state.go

```diff
@@ -71,47 +83,58 @@ next:
 
 	switch {
 	case l.ch == '\n':
-		// semicolon insertion
+		// auto semicolon insertion
 		if l.insertSemi {
 			l.emit(token.SEMICOLON)
 			return
 		}
 
+		// ignore all space
 		consumeAllSpace(l)
+
 	case unicode.IsSpace(l.ch):
-		// ignore whitespace
-		consumeSpace(l)
+		if l.insertSemi {
+			// semicolon to be inserted
+			// do not ignore newlines
+			consumeSpace(l)
+		} else {
+			// no semicolon to be inserted
+			// ignore all space runes
+			consumeAllSpace(l)
+		}
```

Comment:
There is no `fallthrough`- I think you can `break` on one of them and do the other one outside `else`.

Reply:
Done.